scripts/scons_plugin/systemdeps.py

```diff
@@ -0,0 +1,100 @@
+import os
+
+def GeneratePkgConfig(env, build_dir, filename, prefix, libdir, name, desc, url, version):
+    target = os.path.join(build_dir, filename)
+
+    dep_list = env.get('PKG_CONFIG_DEPS', [])
+
+    lib_list = \
+      ['roc'] + \
+      env.get('PKG_CONFIG_LIBS', [])
+
+    incdir_list = \
+      [prefix+'/include'] + \
+      env.get('PKG_CONFIG_CPPPATH', [])
+
+    libdir_list = \
+      [libdir] + \
+      env.get('PKG_CONFIG_LIBPATH', [])
+
+    src = """prefix=%(prefix)s
+exec_prefix=${prefix}
+
+Name: %(name)s
+Requires: %(deps)s
+Version: %(version)s
+Description: %(desc)s
+URL: %(url)s
+
+Libs: %(libdirs)s %(libs)s
+Cflags: %(incdirs)s
+""" % { 'prefix': prefix,
+        'name': name,
+        'desc': desc,
+        'url': url,
+        'version': version,
+        'deps': ' '.join(dep_list),
+        'libdirs': ' '.join(['-L'+s for s in libdir_list]),
+        'incdirs': ' '.join(['-I'+s for s in incdir_list]),
+        'libs': ' '.join(['-l'+s for s in lib_list]),
+        }
+
+    def write_file(target, source, env):
+        f = open(target[0].path, 'w')
+        f.write(src)
+        f.close()
+
+    env.Command(target, [], [
+        env.Action(write_file, env.PrettyCommand('GEN', env.File(target).path, 'purple')),
+        ])
+
+    return env.File(target)
+
+# Add system dependency manually. Used as fallback when AddPkgConfigDependency did not
+# work or can't be used.
+# If libs arg is provided, libs are added to linker options.
+# If prefix arg is provided, and it defines a dir which has include and lib subdirs,
+# those subdirs are added to the headers and library search paths.
+# This function also automatically informs GeneratePkgConfig that correspondings
+# paths and libs should be added to .pc file.
+def AddManualDependency(env, libs=[], prefix=None):
+    # append values to env[<name>] and env[PKG_CONFIG_<name>]
+    def append_var(name, values):
+        for var in [name, 'PKG_CONFIG_'+name]:
+            if var not in env.Dictionary():
+                env[var] = []
+            env.AppendUnique(**{var: values})
```

Comment:
is it different from `env.AppendUnique(var=values)`?

Reply:
In

```
env.AppendUnique(var=values)
```

variable name is string `"var"`

And in

```
env.AppendUnique(**{var: values})
```

variable name is value of variable var.

In this regard, AppendUnique works similar to format().